lib/openapi_parser/schema_validators/string_validator.rb

```diff
@@ -101,5 +82,23 @@ def validate_date_format(value, schema)
 
         return [value, nil]
       end
+
+      def validate_datetime_format(value, schema)
+        return [value, nil] unless schema.format == 'date-time'
+
+        begin
+          if @datetime_coerce_class.nil?
+            # validate only
+            Time.iso8601(value)
```

Comment:
Time.iso8601 pass format which is invalid format in RFC3339
Ruby's ios8601 method use xmlschema's date-time format.
And this format allow date time format without time zone.
https://www.w3schools.com/xml/schema_dtypes_date.asp

On the other hand, RFC3339 require time-offset so `2020-01-01T00:00:00`(without time zone) is invalid format (we need time-offset like `2020-01-01T00:00:00Z`)

```
irb(main):021:0> Time.iso8601("2020-01-01T00:00:00Z")
=> 2020-01-01 00:00:00 UTC
irb(main):022:0> Time.iso8601("2020-01-01T00:00:00")
=> 2020-01-01 00:00:00 +0900
irb(main):023:0> DateTime.rfc3339("2020-01-01T00:00:00Z")
=> #<DateTime: 2020-01-01T00:00:00+00:00 ((2458850j,0s,0n),+0s,2299161j)>
irb(main):024:0> DateTime.rfc3339("2020-01-01T00:00:00")
Traceback (most recent call last):
        5: from /home/ota42y/.rbenv/versions/3.0.0/bin/irb:23:in `<main>'
        4: from /home/ota42y/.rbenv/versions/3.0.0/bin/irb:23:in `load'
        3: from /home/ota42y/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/gems/irb-1.3.0/exe/irb:11:in `<top (required)>'
        2: from (irb):24:in `<main>'
        1: from (irb):24:in `rfc3339'
Date::Error (invalid date)
```

OpenAPI 3 use RFC3339 format so please use DateTime.rfc3339 to validate date

Reply:
Thank you for the review and detailed information!
I fixed it :ok_hand: